### PR TITLE
feat: upgrade to Angular 13

### LINKED ISSUE #13916: Upgrade to Angular 13
### Checklist:
- [x] upgrade Angular to the latest version, see [Angular Advanced upgrade checklist](https://update.angular.io/). Changes:
  - Angular [CHANGELOG](https://github.com/angular/angular/blob/master/CHANGELOG.md) and [blogpost](http://goo.gle/angular-v13)
  - Angular CLI [CHANGELOG](https://github.com/angular/angular-cli/blob/master/CHANGELOG.md)
    - Note: Angular CLI repo covers many devkit and schematics: @angular/cli, @schematics/angular, @angular-devkit/build-angular, @angular-devkit/build-webpack, @angular-devkit/core, @angular-devkit/schematics, @ngtools/webpack
       - [x] if not all above packages were upgraded automatically via ng update @angular/cli@13, upgrade them separately
  - ng-packgr [CHANGELOG](https://github.com/ng-packagr/ng-packagr/blob/master/CHANGELOG.md)
- [x] upgrade major Angular-related dependencies (using `ng update` command! to benefit from automated migration)  **to be compatible with the latest Angular**:
  - [x] @nguniversal/express-engine  [CHANLGELOG](https://github.com/angular/universal/blob/master/CHANGELOG.md) 
  - [x] TypeScript [CHANGELOG](https://github.com/microsoft/TypeScript/releases)
  - [x] Ngrx [CHANGELOG](https://github.com/ngrx/platform/blob/master/CHANGELOG.md)
      - note: breaking change was applied within schematics on update command (removing `@Effect()` decorator)
  - [x] ng-bootstrap [CHANGELOG](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CHANGELOG.md), [npm](https://www.npmjs.com/package/@ng-bootstrap/ng-bootstrap) 
    - :warning:  Check if NG13 is already supported [CHANGELOG](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CHANGELOG.md)
  - [x] ng-select [CHANGELOG](https://github.com/ng-select/ng-select/blob/master/CHANGELOG.md)
    - :warning:  Check if NG13 is already supported [CHANGELOG](https://github.com/ng-select/ng-select/releases?q=angular+13&expanded=true)
  - [x] angular-oauth2-oidc [CHANGELOG](https://github.com/manfredsteyer/angular-oauth2-oidc/blob/master/CHANGELOG.md) - be cautious when bumping to new major, double check consequences and breaking changes
- [x] upgrade major dev-dependencies related to Angular:
   - [x] @angular-builders/custom-webpack [repo](https://github.com/just-jeb/angular-builders)
   - [x] @angular-eslint/angular-eslint [CHANGELOG](https://github.com/angular-eslint/angular-eslint/blob/master/CHANGELOG.md)
       - blocked by: https://github.com/angular-eslint/angular-eslint/issues/715
- [x] **optional** upgrade 3rd party dependencies not related to Angular:
   - RxJs [CHANGELOG](https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md), [summary of changes between 6.x. and 7.x](https://rxjs.dev/6-to-7-change-summary) 
      - decision: due to time matter rxjs update effort will be separate topic #15354
   - [x] bootstrap [CHANGELOG](https://github.com/twbs/bootstrap/releases) - bump only to latest minor+patch, but not major to avoid unnecessary styling breaking changes
  - [x] i18next [CHANGELOG](https://github.com/i18next/i18next/blob/master/CHANGELOG.md) & [Migration guide](https://www.i18next.com/misc/migration-guide#v20.x.x-to-v21.0.0) - be cautious when bumping to new major, double check consequences and breaking changes
    - [x] i18next-http-backend [NO CHANGELOG](https://github.com/i18next/i18next-http-backend) - bump to be in line with i18next
 - [x] when the main package.json is ready, run `yarn config:update --bump-versions` and make sure that we:
   - [x] bump peer-dependencies in all `package.json` files of our libraries 
   - [x] bump peer-dependencies in the `dependencies.json` file used for validation of versions
- [x] apply the ng13 migrations (in `test.ts` and `ng-package.json` like [here](https://github.com/SAP/spartacus/pull/14323/commits/ebeb2b19944254d53f13a674a42a83b21075e8cc)) in new entry points in 5.0 in:
   - [x] in cart lib, when it's ready // @WeizhengSap @plabadie 
   - [x] in checkout lib, when it's ready // @znikola @bgambocjaviniar 
   - [x] in epd-visualisations lib (should be added in the meantime to 4.2)
- [x] search for any mentions about previous version of angular in our repo and make sure to update them to newest
  - [x] `ANGULAR_CLI_VERSION` variable in `scripts/install/config.default.sh` 

### Issues found (checked = fixed):
 - [x]  `@angular-builders/dev-server": "^7.3.1` which is in our package.json doesn't have an equivalent 13 version published. Do we need it? Can we drop it from our package.json?
- [x] `ClientRect` changed signature in TS 4.4 and is no longer valid in Spartacus `PositioningService`
   -  Fix: replaced `ClientRect` with a custom interface
- [x] `yarn start` produces error: `Generating browser application bundles (phase: building)...Unknown error from PostCSS plugin. Your current PostCSS version is 8.3.0, but postcss-preset-env uses 7.0.35. Perhaps this is the source of the error below`
   - related Angular issue: angular/angular-cli#20967
   - Fix: aligned our `.browserlistrc` file with latest Angular fresh app's `.browserlistsrc`
- [x] `ng build core --configuration production` produces error: `The builder requires a target`
   - Related Angular issue: angular/angular-cli#22152
   - Fix: pass target in the 3rd parameter of the scheduleBuilder()
- [x] `yarn start` produces runtime console.warn<details><summary>WARNING:</summary>`[webpack-dev-server] WARNING in ./projects/storefrontapp/src/styles/lib-cart.scss (./projects/storefrontapp/src/styles/lib-cart.scss.webpack[javascript/auto]!=!./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[4].rules[0].oneOf[1].use[1]!./node_modules/@angular-devkit/build-angular/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[4].rules[0].oneOf[1].use[2]!./node_modules/resolve-url-loader/index.js??ruleSet[1].rules[4].rules[1].use[0]!./node_modules/@angular-devkit/build-angular/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[4].rules[1].use[1]!./projects/storefrontapp/src/styles/lib-cart.scss) Module Warning (from ./node_modules/@angular-devkit/build-angular/node_modules/postcss-loader/dist/cjs.js): Warning (11060:3) postcss-preset-env: start value has mixed support, consider using flex-start instead`</details>
   - Fix: replace usages of `justify-content: start` to `justify-content: flex-start` and test if looks the same
- [x] unit tests of core / storefront fail `TOTAL: 13 FAILED, 2535 SUCCESS`
   - [x] `JavaRegExpConverter` - `/pattern/gimu` regex literal in unit tests is changed to `/pattern/gim`( the`u` flag is dropped somehow). 
       - workaround: change regex literal to regex object `new Regex('pattern', 'gimu')`
       - fix: set `"target": "es2015"` in `tsconfig.spec.json`
   - [x] `GlobalMessageEffect` - our previous hack to mock the `delay` rxjs operator no longer works
      - fix: refactor unit tests to use rxjs TestScheduler + apply timings (ms) into marbles
   - [x] `OccProductAdapter` - `TypeError: Cannot destructure undefined` / `TypeError: Cannot destructure property 'queryParams' of 'undefined' as it is undefined.`
      - fix: don't destructure undefined object `queryParams` in the mocked `MockOccEndpointsService`
- [x] unit tests of other libs don't have enough coverage 🤔 `Coverage for branches (54.22%) does not meet global threshold (70%)`
         - fix: there's possibility that switching to new reporter `karma-coverage` causes more strict approach, which forced us to supply few unit tests to float above treshold
 - [x] [16 e2e failed](https://dashboard.cypress.io/projects/k3nmep/runs/27880/overview?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D)
        - fix: 2 fixes were applied. 1. for pluralization in translation due to BC in `i18next` lib. 2. Due to routing changes with required property for routerLink we had to supply default param for `PaginationComponent.pageRoute` property 
  - after upgrading rxjs to version 7.0.1 and jasmine-marbles to 0.9.1 there is 72 tests failing. Most of issues come from error: Error: No test scheduler initialized
    - decision: decision: due to time matter jasmine-marbles update effort will be done in separate topic as part of rxjs update effort #15354
- [x] 11 failing tests in `projects/core` due to compilation problems in es6 mode with `--code-coverage` flag included in command. The issue is related to dependencies not being injected in to the inherited constructor of the class extending base class, causing unit tests to fail. [Related comment](https://github.com/angular/angular-cli/issues/14860#issuecomment-1017424113)
      - fix: bumped Angular to minor 13.2 as it contained fix for issue linked above
- [x] Unable to run unit tests for all feature-libs/XYZ/schematics. 🤔  `Cannot find module '@angular/compiler' from '../../projects/schematics/src/shared/utils/file-utils.ts'`. Investigation required. 
      - Fix: due to new module format with NG13 (.mjs) we need to use an external library [jest-preset-angular](https://github.com/thymikee/jest-preset-angular) to keep Jest running unit tests properly
- [x] When running `ng add @spartacus/schematics` in a fresh app, the error occurs: `require() of ES Module <path-to-app>/node_modules/@angular/compiler/fesm2015/compiler.mjs not supported.
Instead change the require of <path-to-app>/node_modules/@angular/compiler/fesm2015/compiler.mjs to a dynamic import() which is available in all CommonJS modules.`
     - Proposed fix: in our schematics change the import of `@angular/compiler` to a dynamic import (PR https://github.com/SAP/spartacus/pull/15273)
- [x] Problem with fresh ng13 app with added Spartacus 5.0 (via local verdaccio): `(from ./node_modules/sass-loader/dist/cjs.js):
SassError: Can't find stylesheet to import.
   |
1 │ @import "@spartacus/cart";
   |
  src/styles/spartacus/cart.scss 1:9  root stylesheet` 
  - fix: [package.json file adjustment to include index.scss files](https://github.com/SAP/spartacus/pull/14323/commits/a8acb042002732c6d765900465227b2a58c89707) due to Angular changing way of bundling packages so we need to [manually add pre-compiled styling files to package.json](https://angular.io/guide/creating-libraries#managing-assets-in-a-library)
- [x] Problem with schematics while adding Spartacus to fresh app ng13 using schematics from our branch (`feature/GH-13916-schematics-refactor`): `mystore-ng-13/node_modules/semver/index.js'. Please verify that the package.json has a valid "main" entry`:
  - reason: The `"main"` entry point filename in `package.json` of `semver` package is being **changed** from `index.js` to `semver.js` **during the execution** of the command `ng add @spartacus/schematics` . It's because `@angular/cli@13` uses new major version of `semver` (semver@7.3.5) dependency, different than `@spartacus/schematics` (semver@6.3.0) and those versions have different entry point filename ([see changelog of `semver@7.0.0`](https://github.com/npm/node-semver/blob/main/CHANGELOG.md#700)). It seems that Node resolves the entry point filename of `semver` using the `package.json` from the start of executing the command `ng add ...` (when it's `"main": "index.js"`), but not from the `package.json` after `@spartacus/schematics` is yarn-installed (when it' `"main": "semver.js"`).    
  - workaround: bump `semver` from 6.3.0 to 7.3.5 in our schematics. This avoids changing the field `"main": "index.js"` of `node_modules/semver/pacakge.json` during the execution of `ng add @spartacus/schematics` in customer's app
  - note for future Angular upgrades: if Angular CLI bumps again `semver` package in the future in their internal `dependencies`, we will need to bump it again in `dependencies` of our schematics
- [x] update file [creating-lib.md](https://github.com/SAP/spartacus/blob/2bc2dfbedb46a5c2b83fdc90a96c0a5688366742/docs/libs/creating-lib.md), because creating lib has slightly changed (e.g. you don't need umd modules declarations in ng-package.json. you need new "scss" section in package.json files, etc.)
- [x] https://github.com/SAP/spartacus/issues/15457
   - not an issue. it happens only in devmode and doesn't affect the JS code-splitting efforts

   
### Investigate consequences of the following breaking changes:
- [x] Angular
  - [x] "You can now disable the navigation of a routerLink by passing undefined and null. Previously the routerLink directive used to accept these two values as equivalent to an empty string." 
     - Spartacus: RoutingService.go, RoutingService.getUrl, SemanticPathService.get, SemanticPathService.transform, UrlPipe.transform; places that have might have null/undefined in the [routerLink], e.g. due to misconfigured routing config

### Reminder for libraries bump when stables get released: (hopefully before we merge)
- [x]  [@nguniversal/express-engine](https://www.npmjs.com/package/@nguniversal/express-engine) (currently latest is 13.0.2)
  - no need to wait for the new version of `@nguniversal` - 13.0.2 suffices. Moreover [Angular maintainers said they won't couple versions of main `@angular` packages with `@nguniversal` except major releases](https://github.com/angular/universal/issues/2527)

### Testing checklist:
- [x] yarn start - app compilation smoke test
- [x] yarn build:libs - libs compilation test
- [x] yarn build - app prod compilation
- [x] run unit tests
- [x] SSR smoke test: homepage, PDP, PLP, faq (with JS disabled in the browser):
  - [x] page renders correctly
  - [x] no console messages
  - [x] no UI flicker after CSR app is bootstrapped in the browser
- [x] lazy loading smoke test: traverse storefront, observe loading scripts in network tab
- [x] a fresh Spartacus app installed via schematics (installation script):
  - [x] smoke test: traverse homepage, PDP, PLP, faq, do checkout
  - [x] lazy loading smoke test - traverse storefront, observe loading scripts in network tab
  - [x] verify the dependencies in the fresh app's package.json are in line with Spartacus repo's package.json
- [x] generate a fresh ng13 app and check its default configs, tsconfig flags, dependency versions, tslib, zone.js, etc. and compare it with our setup
- [x] e2e pass for:
  - [x] core libs
  - [x] integration libs ( EDIT @szczepaniak-michal :  e2e tests are failing for vendor libs, we have contacted the libraries owners)
    - [x] CPQ: need to contact the Team Tiger / @ChristophHi
    - [x] CDS: need to find an owner
    - [x] Digital Payments: need to find an owner
    - [x] EPD Visualisations: need to contact @sam-garland
- [x] [type augmentation](https://sap.github.io/spartacus-docs/type-augmentation/#augmenting-modules) of `Product` from `@spartacus/core` works in the fresh app
 
### Follow up work:
 - [x] Announce changes for the dev team: 
   - (Must) pull develop branch and run yarn
   - (Should) if needed, update your local Node.js version to the latest LTS (v16) https://nodejs.org/en/download/
   - (Should) Upgrade VSCode extension "Angular Language Service" to version 13 image
      - Wait unitl this issue is resolved https://github.com/angular/angular/issues/44148
   - (Info) Build time for subsequent local builds decreased 13% thanks to [caching build artifacts by default](https://angular.io/cli/cache)
     - 13% for `yarn build:libs`
     - 67% for `yarn build` (app prod build)
     - 34% for `yarn start` (app dev server)

### References:
- Past https://github.com/SAP/spartacus/issues/12058
- Past https://github.com/SAP/spartacus/pull/12926

### PR DESCRIPTION
**Pull request by Platonn**
Main changes:
- bump `@angular/` packages from v12.0 to v13.2
- bump `typescript` from v4.2 to v4.4 
- bump relative packages (e.g. `@nguniversal`, `@ngrx`, `ng-select` etc.) to the highest version compatible with ng13.2
- bump `i18next` from v20 to v21
- bump css `bootstrap` from v4.3 to v4.6

Consequence changes:
- drop section `umdModuleIds` from `ng-package.json` files - it's not needed since ng13
- add "exports/sass" section in `package.json` files of libraries - needed since ng13 introduced new "Angular Package Format"
- bump required Node.js version in our repo to `^14.15.0 || >=16.10.0` - same as in ng13 (but excluding Node12 because on 2022-04-30 Node.js v12 will become EOL)
  - use Node 16 (instead of Node 12) in CI
- add eslint rules needed after bumping `@angular-eslint` and `eslint`
  - rename variables here and there to conform the `no-shadow` eslint rule
  - change comparisons `==` to `===` here and there to conform `eqeqeq` rule
- bump `jest`, bump `ts-jest` and add `jest-preset-angular` - needed since ng13 is shipped only as ES Modules (no CommonJS)
  - btw. renamed files `jest.ts` to `test-jest.js`
- configure `destroyAfterEach: false` for unit tests of existing libraries - since ng13 the default is `destroyAfterEach: true`
- use CSS `justify-content: 'flex-start',` instead of `'start'` to avoid to warning since ng13
- bump `semver` package used in our schematics - same as ng13
- import ES Module of `@angular/compiler` in our CommonJS migration-schematics via a workaround function `loadEsmModule()` - since ng13 `@angular/compiler` is shipped only as ES Module
- force downgrade of rxjs in the fresh app from v7 to v6  via installation-schematics  - needed since `ng new` in v13 installs rxjs@7, but Spartacus requires v6
- update `.browserslistrc` for `storefrontapp` project - same as in a fresh ng13 app
- add default value `'.'` of `@Input pageRoute` in `PaginationComponent` - needed since ng13 disables links with empty `routerLink`
- add simple unit tests in some libraries - after upgrading to ng13 the coverage was below the threshold for some libraries
- add new param `target` of `context.scheduleBuilder()` in `augmented-types.ts` builder - needed since ng13
- change NgRx `@Effect()` decorator to `createEffect()` function  - needed since ngrx@13
- update sufixes of plural i18next keys from `_plural` to `_other` - needed since i18next@21

closes #13916

### CODE REVIEW

File: docs/migration/5_0.md [hunk 1]
```diff
@@ -444,8 +533,74 @@ The following keys have been added to `configurator-common.ts`:
 - `[cxFoxus]` directive was removed from `<cx-facet>` in template to support Screen Readers.
 - `(unlock)` directive was removed from `<cx-facet>` in template to support Screen Readers.
 
-### StoreFinderComponent
+### Translation (i18n) changes related to accessibility in variant configuration
```
Comment (Platonn) [inline docs/migration/5_0.md]: Why those changes are part of this PR?

File: feature-libs/checkout/base/components/services/express-checkout.service.ts [hunk 2]
```diff
@@ -86,9 +86,9 @@ export class ExpressCheckoutService {
         }
         return of([supportedDeliveryModesState]).pipe(
           filter(
-            ([supportedDeliveryModesState]) =>
-              !supportedDeliveryModesState.loading &&
-              !!supportedDeliveryModesState.data?.length
+            ([supportedDeliveryModesStateObject]) =>
+              !supportedDeliveryModesStateObject.loading &&
+              !!supportedDeliveryModesStateObject.data?.length
```
Comment (Platonn) [inline feature-libs/checkout/base/components/services/express-checkout.service.ts:91]: Are those changes related to this PR?

File: feature-libs/checkout/scheduled-replenishment/components/checkout-schedule-replenishment-order/checkout-schedule-replenishment-order.component.html [hunk 3]
```diff
@@ -18,7 +18,7 @@ <h5 class="cx-order-replenishment-header">
       aria-checked="true"
       (change)="changeOrderType(type.value)"
       [value]="type.value"
-      [checked]="type.value == (selectedOrderType$ | async)"
+      [checked]="type.value === (selectedOrderType$ | async)"
```
Comment (Platonn) [inline feature-libs/checkout/scheduled-replenishment/components/checkout-schedule-replenishment-order/checkout-schedule-replenishment-order.component.html:21]: Is this change related to this PR?

File: feature-libs/order/components/return-request-list/order-return-request-list.component.spec.ts [hunk 4]
```diff
@@ -80,14 +90,35 @@ describe('OrderReturnRequestListComponent', () => {
...
+    component.returnRequests$
+      .subscribe((value) => {
+        returns = value;
+      })
+      .unsubscribe();
+
+    expect(returns).toEqual({});
+    expect(component.sortType).toBe(undefined);
+  });
+
```
Comment (Platonn) [inline feature-libs/order/components/return-request-list/order-return-request-list.component.spec.ts:107]: are changes in this file related to this PR?
(similar concern for other spec files)

File: feature-libs/organization/administration/core/store/effects/budget.effect.ts [hunk 5]
```diff
@@ -11,117 +11,121 @@ import { HttpErrorResponse } from '@angular/common/http';
...
+        this.budgetConnector.create(payload.userId, payload.budget).pipe(
           switchMap((data) => [
-            new BudgetActions.UpdateBudgetSuccess(data),
+            new BudgetActions.CreateBudgetSuccess(data),
             new OrganizationActions.OrganizationClearData(),
           ]),
           catchError((error: HttpErrorResponse) =>
             from([
-              new BudgetActions.UpdateBudgetFail({
+              new BudgetActions.CreateBudgetFail({
```
Comment (Platonn) [inline feature-libs/organization/administration/core/store/effects/budget.effect.ts:90]: Are those changes of NgRx action names related to this PR?

File: feature-libs/product-configurator/jest.schematics.config.js [hunk 6]
```diff
@@ -11,10 +12,9 @@ module.exports = {
       tsconfig: '<rootDir>/tsconfig.schematics.json',
     },
   },
-
   collectCoverage: false,
   coverageReporters: ['json', 'lcov', 'text', 'clover'],
-  coverageDirectory: '<rootDir>/../../coverage/storefinder/schematics',
+  coverageDirectory: '<rootDir>/../../coverage/product-configurator/schematics',
```
Comment (Platonn) [inline feature-libs/product-configurator/jest.schematics.config.js:17]: nice fix btw

File: feature-libs/product-configurator/rulebased/core/facade/configurator-commons.service.spec.ts [hunk 7]
```diff
@@ -102,9 +102,9 @@ class MockconfiguratorUtilsService {
     return configId !== undefined && configId.length !== 0;
   }
   getConfigurationFromState(
-    configurationState: StateUtils.ProcessesLoaderState<Configurator.Configuration>
+    configurationLoaderState: StateUtils.ProcessesLoaderState<Configurator.Configuration>
```
Comment (Platonn) [inline feature-libs/product-configurator/rulebased/core/facade/configurator-commons.service.spec.ts:105]: I understand this change is related to `no-shadow` rule of the linter

File: feature-libs/tracking/personalization/root/personalization-root.module.spec.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  it('initializes', () => {
+    const module = TestBed.inject(PersonalizationRootModule);
+    expect(module).toBeTruthy();
+  });
+
+  it('defaultPersonalizationComponentsConfig', () => {
+    const result = defaultPersonalizationComponentsConfig();
+    expect(result).toEqual(MockCmsConfig);
+  });
+});
```
Comment (Platonn) [inline feature-libs/tracking/personalization/root/personalization-root.module.spec.ts:33]: Are changes in this file related to this PR?

File: feature-libs/checkout/scheduled-replenishment/components/checkout-schedule-replenishment-order/checkout-schedule-replenishment-order.component.html (hunk 3 above)
Comment (Platonn) [reply feature-libs/checkout/scheduled-replenishment/components/checkout-schedule-replenishment-order/checkout-schedule-replenishment-order.component.html:21]: I guess yes and it's related to the upgraded linter.

File: feature-libs/checkout/base/components/services/express-checkout.service.ts (hunk 2 above)
Comment (Platonn) [reply feature-libs/checkout/base/components/services/express-checkout.service.ts:91]: I guess yes and it's related to the upgraded linter (`no-shadow` rule)

File: feature-libs/order/components/order-history/order-history.component.spec.ts [hunk 9]
```diff
@@ -220,17 +221,46 @@ describe('OrderHistoryComponent', () => {
...
+
+    component.orders$
+      .subscribe((value) => {
+        orders = value;
+      })
+      .unsubscribe();
+
+    expect(orders).toEqual(undefined);
+
+    expect(component.sortType).toBe(undefined);
```
Comment (Platonn) [inline feature-libs/order/components/order-history/order-history.component.spec.ts:239]: Are those changes related to this PR?

File: projects/core/src/anonymous-consents/store/selectors/anonymous-consent-templates.selectors.ts [hunk 10]
```diff
@@ -15,7 +15,8 @@ export const getAnonymousConsentTemplatesValue: MemoizedSelector<
   ConsentTemplate[]
 > = createSelector(
   getAnonymousConsentTemplatesState,
-  StateUtils.loaderValueSelector
+  (state: StateUtils.LoaderState<ConsentTemplate[]>) =>
+    StateUtils.loaderValueSelector(state)
```
Comment (Platonn) [inline projects/core/src/anonymous-consents/store/selectors/anonymous-consent-templates.selectors.ts:19]: Why do we need to do this change in this PR?

File: projects/core/src/global-message/store/effects/global-message.effect.spec.ts [hunk 11]
```diff
@@ -18,17 +17,6 @@ import {
...
-  const oldProp: Function = obj[prop];
-  Object.defineProperty(obj, prop, {
-    configurable: true,
-    enumerable: true,
-    value: oldProp,
-    writable: true,
-  });
-  return spyOn(obj, prop);
-}
-
```
Comment (Platonn) [inline projects/core/src/global-message/store/effects/global-message.effect.spec.ts:31]: Just for reference:
This trick for mocking rxjs `delay` operator no longer works since ng13.
So now we improved the rxjs-marbles in unit tests to take into the consideration the actual the real delays (e.g. 3000ms or 10 000ms)

File: projects/storefrontlib/cms-components/navigation/search-box/search-box.component.spec.ts [hunk 12]
```diff
@@ -46,7 +46,9 @@ class MockCmsComponentData {
   name: 'cxUrl',
 })
 class MockUrlPipe implements PipeTransform {
-  transform(): any {}
+  transform(): any {
+    return ['test', 'url'];
+  }
```
Comment (Platonn) [inline projects/storefrontlib/cms-components/navigation/search-box/search-box.component.spec.ts:51]: Is this change related to this PR?

File: feature-libs/tracking/personalization/root/personalization-root.module.spec.ts (hunk 8 above)
Comment (psmul) [reply feature-libs/tracking/personalization/root/personalization-root.module.spec.ts:33]: At first glance, I thought this was omitted file during one of the merges from develop, but then I reminded myself that this was an effort to bump unit tests threshold on pipeline done by @mszczepaniak

File: feature-libs/order/components/order-history/order-history.component.spec.ts (hunk 9 above)
Comment (psmul) [reply feature-libs/order/components/order-history/order-history.component.spec.ts:239]: Effort done by @mszczepaniak to bump unit test coverage above defined threshold for pipeline

File: feature-libs/order/components/return-request-list/order-return-request-list.component.spec.ts (hunk 4 above)
Comment (psmul) [reply feature-libs/order/components/return-request-list/order-return-request-list.component.spec.ts:107]: Effort done by @mszczepaniak to bump unit test coverage above the defined threshold for pipeline

**Review by psmul — APPROVED** [review]
